src/nile/deployments.py

```diff
@@ -28,6 +28,58 @@ def register(address, abi, network, alias):
         fp.write("\n")
 
 
+def update_abi(address_or_alias, abi, network):
+    """
+    Update the ABI for an existing deployment that matches an identifier.
+
+    If address_or_alias is an int, address is assumed.
+
+    If address_or_alias is a str, alias is assumed.
+    """
+    file = f"{network}.{DEPLOYMENTS_FILENAME}"
+
+    if not os.path.exists(file):
+        raise Exception(f"{file} does not exist")
+
+    with open(file, "r") as fp:
+        lines = fp.readlines()
+
+    found = False
+    for i in range(len(lines)):
+        line = lines[i].strip().split(":")
+
+        current_address_hex = line[0]
+
+        current_aliases = None
+        length = len(line)
+        if length > 2:
+            current_aliases = line[-(length - 2) :]
+
+        if (
+            type(address_or_alias) is not int and address_or_alias in current_aliases
+        ) or (
+            type(address_or_alias) is int
+            and address_or_alias == normalize_number(current_address_hex)
+        ):
+            logging.info(f"📦 Updating deployment {address_or_alias} in {file}")
+
+            replacement = f"{current_address_hex}:{abi}"
+            if current_aliases is not None:
+                replacement += ":" + ":".join(str(x) for x in current_aliases)
+            replacement += "\n"
+
+            lines[i] = replacement
+
+            found = True
+            break
```

Comment:
We can use the same pattern as in `load` method here I think:
```suggestion
        [address, abi, *aliases] = line.strip().split(":")
        address = normalize_number(address)
        identifiers = [address]
        
        if type(address_or_alias) is not int:
            identifiers = alias
        
        if address_or_alias in identifiers:
            logging.info(f"📦 Updating deployment {address_or_alias} in {file}")
            
            replacement = f"{address}:{abi}"
            if aliases is not None:
                replacement += ":" + ":".join(str(x) for x in aliases)
            replacement += "\n"
            lines[i] = replacement
            found = True
            break
```